packages/resolvers/glob/src/GlobResolver.js

```diff
@@ -33,26 +55,65 @@ export default (new Resolver({
     }
 
     if (error) {
-      throw new ThrowableDiagnostic({
-        diagnostic: {
-          message: error,
-          codeFrames: dependency.loc
-            ? [
-                {
-                  codeHighlights: [
-                    {
-                      start: dependency.loc.start,
-                      end: dependency.loc.end,
-                    },
-                  ],
-                },
-              ]
-            : undefined,
-        },
+      throw errorToThrowableDiagnostic(error, dependency);
+    }
+
+    // if the specifier does not start with /, ~, or . then it's not a path but package-ish - we resolve
+    // the package first, and then append the rest of the path
+    if (!/^[/~.]/.test(specifier)) {
+      specifier = path.normalize(specifier);
```

Comment:
I _think_ globs are supposed to always use `/` as a separator, even on Windows. See https://github.com/micromatch/micromatch#backslashes. So this might not be needed.